tests/rptest/utils/mode_checks.py

```diff
@@ -33,33 +34,35 @@ def cleanup_on_early_exit(caller):
         allocate_and_free(test_context.cluster, caller.logger)
 
 
-def skip_debug_mode(func):
+def is_debug_mode():
+    return os.environ.get('BUILD_TYPE', None) == 'debug'
+
+
+def skip_debug_mode(*args, **kwargs):
     """
-    Decorator applied to a test class method. The property `debug_mode` should be present
-    on the object.
+    Test method decorator which signals to the test runner to ignore a given test.
+
+    Example::
+
+        When no parameters are provided to the @ignore decorator, ignore all parametrizations of the test function
+
+        @skip_debug_mode  # Ignore all parametrizations
+        @parametrize(x=1, y=0)
+        @parametrize(x=2, y=3)
+        def the_test(...):
+            ...
+
+    Example::
 
-    If set to true, the wrapped function call is skipped, and a cleanup action
-    is performed instead.
+        If parameters are supplied to the @skip_debug_mode decorator, only skip the parametrization with matching parameter(s)
 
-    If set to false, the wrapped function (usually a test case) is called.
+        @skip_debug_mode(x=2, y=3)
+        @parametrize(x=1, y=0)  # This test will run as usual
+        @parametrize(x=2, y=3)  # This test will be ignored
+        def the_test(...):
+            ...
     """
-    @functools.wraps(func)
-    def f(*args, **kwargs):
-        assert args, 'skip_debug_mode must be placed on a test method in a class'
-
-        caller = args[0]
-
-        assert hasattr(
-            caller, 'debug_mode'
-        ), 'skip_debug_mode called on object which does not have debug_mode attribute'
-        assert hasattr(
-            caller,
-            'logger'), 'skip_debug_mode called on object which has no logger'
-        if caller.debug_mode:
-            caller.logger.info(
-                "Skipping test in debug mode (requires release build)")
-            cleanup_on_early_exit(caller)
-            return None
-        return func(*args, **kwargs)
-
-    return f
+    if is_debug_mode():
+        return ignore(args, kwargs)
+    else:
+        return args[0]
```

Comment:
We should call `allocate_and_free` and the `early_exit_hook` that a few tests use to cleanup when skipping tests.

Reply:
We are not allocating any resources for ignored tests. Similar to the ducktape's `@ignore` hook. Why would we need to call the above functions? Per my understanding, previous code would actually initialize the test bed so we had to use the resources to avoid warnings. The updated code doesn't do any of these.

Reply:
Yeah, I see if we [ignore the test](https://github.com/confluentinc/ducktape/blob/master/ducktape/tests/runner_client.py#L206) we skip out on the setup  which [runs later](https://github.com/confluentinc/ducktape/blob/master/ducktape/tests/runner_client.py#L233), so this should be fine.